pkg/provisioner/ironic/ironic.go

```diff
@@ -907,10 +910,10 @@ func (p *ironicProvisioner) InspectHardware(data provisioner.InspectData, restar
 		if gophercloud.ResponseCodeIs(err, 404) {
 			// The node has just been enrolled, inspection hasn't been started yet.
 			result, started, err = p.startInspection(data, ironicNode)
-			return
+			return result, started, details, err
 		}
-		result, err = transientError(fmt.Errorf("failed to retrieve hardware introspection data: %w", err))
-		return
+		result, err = transientError(errors.Wrap(err, "failed to retrieve hardware introspection data"))
```

Comment:
The change to errors.Wrap is invalid.

Reply:
For context, github.com/pkg/errors is deprecated and everything should be moving to `fmt.Printf()`. `Wrap()` is not and never has been part of the stdlib `errors` package. So this is the opposite of the way we want to go.

Reply:
indeed, good catch, fixed